server/src/LspServer.ts

```diff
@@ -149,25 +134,74 @@ export class LspServer {
       // Register for all configuration changes.
       await this.connection.client.register(DidChangeConfigurationNotification.type, undefined);
     }
+
+    const profileResult = this.dbtProfileCreator.createDbtProfile();
+    const contextInfo = profileResult.isOk()
+      ? { type: profileResult.value.type, method: profileResult.value.method }
+      : { type: profileResult.error.type, method: profileResult.error.method };
+
+    if (profileResult.isOk()) {
+      void this.prepareDestination(profileResult.value);
+    } else {
+      this.showPrepareDestinationWarning(profileResult.error.message);
+      this.contextInitializedDeferred.resolve();
+    }
+
+    await Promise.all([this.prepareRpcServer(), this.contextInitializedDeferred.promise]);
```

Comment:
Instead of creating promise can we call prepareDestination here?
```
await Promise.all([this.prepareRpcServer(), this.prepareDestination(profileResult)]);
prepareDestination(profileResult) {
    if (profileResult.isOk()) {
       ...
    } else {
      this.showPrepareDestinationWarning(profileResult.error.message);
    }
}
```